### PR TITLE
f-no_cron_city_scoped_gate

### PR DESCRIPTION
**Pull request by Sidarthapati**
# Pull Request

## Summary
Adds a **per-city gate** (`city_fleet_config.meeting_mode`) that turns M6 van meeting-points on/off. In `HUB_RETURN` cities (no van/meeting-point infra) the DA periodically returns to the hub — the hub is the rendezvous — instead of meeting a van, reusing the **same M5 hard-constraint engine and custody chain, minus M6 routing**. Also introduces dedicated M4 custody states so a hub-return parcel no longer masquerades as a van state.

---

## What Changed
- **The gate:** `city_fleet_config.meeting_mode` (`VAN_MEETING` default | `HUB_RETURN`) + `hub_return_interval_minutes` (migration `routing/V6_15`); shared enum `common.domain.MeetingMode`. Ops flips a city via `PUT /routing/fleet/{cityId}`; a `VAN_MEETING` invariant clears any stale interval on flip-back.
- **M6 skips van routing for HUB_RETURN:** `RoutePlanningServiceImpl.plan()` branches to the new `HubReturnScheduleService`, which writes an auto-APPROVED degenerate plan + one `da_cron_schedule` per DA (hub coords, periodic slots) and emits `DA_CRON_SCHEDULED` — no CVRP / set-cover / shuttle. Binding consumers (`HubFeedConsumer` / `DaFeedConsumer`) skip HUB_RETURN cities.
- **M5 pickup feasibility unchanged:** `DispatchServiceImpl.buildFeasibility`/`activeMeetingTime` are coordinate-agnostic — hub coords + periodic times flow through the existing `da_cron_assignment` path. `DaTaskServiceImpl` roll-forward advances a periodic cron to its next slot (COMPLETED only on the final slot) so the hard constraint holds across multiple daily returns.
- **Delivery-from-hub:** new dispatch `HubDeliveryFeedConsumer` assigns `ParcelSortedForDeliveryEvent` to the territory DA (real drop coords via new `common.port.ShipmentLocationPort` + `ShipmentLocationAdapter`), **for HUB_RETURN cities only** — non-gated cities stay on M6.
- **Mode-appropriate DA endpoints:** `POST /dispatch/da/{daId}/tasks/{taskId}/hub-handoff` (pickup drop at hub) + `/hub-collect` (delivery collect at hub), vs the van `/van-handoff` + `/drop-collected`. A 409 guard rejects hub endpoints in a VAN_MEETING city.
- **Dedicated M4 custody states:** `RETURNED_TO_HUB`, `HUB_DELIVERY_ASSIGNED`, `COLLECTED_FROM_HUB` (common `ShipmentState` enum + **`V4_22` Postgres `ALTER TYPE ... ADD VALUE`** migration — the Java enum alone silently failed transitions without the DB type change). `TransitionRegistry` edges; `DaEventsConsumer` maps `HUB_RETURN_HANDOFF_COMPLETED → RETURNED_TO_HUB` and mode-resolves `DROP_ASSIGNED`/`DROP_COLLECTED` to the hub states; customer-visible labels for all three.
- **Cross-module port:** `common.port.CityMeetingModePort` — routing `CityMeetingModeAdapter` is `@Primary`; dispatch (`DispatchConfig`) and orders (`MeetingModeConfig`) ship `@ConditionalOnMissingBean` VAN_MEETING fallbacks so their isolated test contexts boot without depending on routing.
- **M8 scan seams (best-effort, non-blocking):** `HubScanSeamProducer` emits `HUB_ORIGIN_IN` (DA→hub drop) and `HUB_DEST_OUT` (hub→DA collect), gated on HUB_RETURN.

---

## Why This Change?
Some of the 5 cities have no van/meeting-point infrastructure, but they still need the full logistics chain — same M5 feasibility, same custody — with the **hub as the rendezvous** instead of a van loop. A per-city config gate lets ops flip a city into HUB_RETURN without introducing new branches in M5's hard-constraint engine (the feasibility path is coordinate-agnostic, so hub coords + a periodic cadence "just work"). Dedicated M4 states keep the parcel's status honest: previously hub-return reused van states like `HANDED_TO_PICKUP_VAN`, which lied about a van that never existed.

---

## Testing
- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

### Test Evidence
Full `mvn clean install` assembles; module suites green.

```
[INFO] Tests run: 87, Failures: 0, Errors: 0, Skipped: 0   -- orders  (incl. DaEventsConsumerTest)
[INFO] Tests run: 59, Failures: 0, Errors: 0, Skipped: 0   -- routing (incl. HubReturnScheduleServiceTest x3)
[INFO] Tests run: 110, Failures: 0, Errors: 0, Skipped: 0  -- dispatch (incl. DaTaskServiceImplTest roll-forward, DA controller hub-endpoint tests)
[INFO] BUILD SUCCESS
```

Edge cases exercised: null/unknown city mode → `VAN_MEETING` fallback; flip-back to VAN clears a stale `hub_return_interval_minutes`; multi-return-per-day roll-forward keeps the hard constraint after the first hub visit; HUB_RETURN vs VAN_MEETING city yield disjoint plans (BLR = 0 vans / cron-at-hub vs DEL = van loops). Live E2E on the demo branch drove both directions (VAN→HUB and HUB→VAN) to `DROPPED`.

---

### Areas Affected
- [x] Backend
- [ ] Frontend
- [x] Routing
- [ ] Infra
- [x] Database

### Modules Affected
- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [x] M4 — Orders
- [x] M5 — Dispatch
- [x] M6 — Routing
- [ ] M7 — Hub
- [x] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions

---

## Risks / Concerns
- **VAN_MEETING cities unchanged** — the gate defaults via the column default; existing van edges/states are untouched, so the 5 current cities regress to no behaviour change.
- **Additive migrations** — `V6_15` (fleet columns) and `V4_22` (`ALTER TYPE shipment_state ADD VALUE`, PG 12+ safe inside Flyway's tx) are both additive; no data backfill.
- **`routing.hub` queue binds `#`** (pre-existing) — under a live real-M7, non-`ParcelSortedForDeliveryEvent` payloads can reach the single-type consumer and land in the DLQ. Tighten to specific routing keys before M7 goes live.
- **Delivery-from-hub idempotency** relies on mode gating so exactly one consumer acts per city (dispatch `HubDeliveryFeedConsumer` for HUB_RETURN vs M6's `HubFeedConsumer` for VAN_MEETING).

---

## Checklist
- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        log.debug("HUB_RETURN delivery assignment for parcel {}: {}", e.parcelId(), result.outcome());
+    }
+
+    /** Representative coordinate for a hex: the mean of its H3 corner vertices (null if not found). */
+    private double[] hexCentroid(UUID cityId, java.time.LocalDate date, UUID hexId) {
+        List<DaTerritoryResponse> territories = gridService.getDaTerritories(cityId, date);
+        for (DaTerritoryResponse t : territories) {
+            for (TerritoryHexResponse hex : t.hexes()) {
+                if (hex.hexId().equals(hexId)) {
+                    return centroidOf(hex.vertices());
```
Comment (Sidarthapati) [inline dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java:98]: Why are we using centroids to attach for deliveries, shouldnt we have the actual lat lon?

File: dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        double[] centroid = hexCentroid(e.cityId(), e.validDate(), e.destinationHexId());
+        if (centroid == null) {
+            log.error("HUB_RETURN delivery parcel {} — destination hex {} not in any active territory for {}; cannot assign",
+                    e.parcelId(), e.destinationHexId(), e.validDate());
+            return;
+        }
+        AssignmentResult result = dispatchService.assignDelivery(
+                e.parcelId(), e.cityId(), centroid[0], centroid[1], e.destinationHexId());
+        log.debug("HUB_RETURN delivery assignment for parcel {}: {}", e.parcelId(), result.outcome());
```
Comment (Sidarthapati) [inline dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java:89]: Why are we using centroids to attach for deliveries, shouldnt we have the actual lat lon? Are we using this in the actual dispatch service too?

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java [hunk 3]
```diff
@@ -76,6 +99,11 @@ public DaTaskView recordVanHandoff(UUID daId, UUID taskId, List<String> parcelSc
             DaTaskView view = save(task);
             recordCronHandoff(daId, task.getOperatingDate(), parcelScans.size());
             daEventProducer.emitVanHandoffCompleted(daId, task.getCityId(), task.getShipmentId());
```
Comment (Sidarthapati) [inline dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: We still have van handoff emit, like emiiting this event is not correct right?  When there is no m6, this should not happen right, logically this is wrong no?

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java [hunk 4]
```diff
@@ -76,6 +99,11 @@ public DaTaskView recordVanHandoff(UUID daId, UUID taskId, List<String> parcelSc
...
+            // M8-SEAM: in HUB_RETURN cities the DA drops collected pickups AT the hub, so this handoff
+            // is the origin-hub inbound scan (advances the shipment to AT_ORIGIN_HUB → M7/M9).
+            if (isHubReturn(task)) {
+                hubScanSeamProducer.hubOriginIn(task.getShipmentId());
+            }
```
Comment (Sidarthapati) [inline dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: Why have we added this staement , there is no such thing as van handoff in a city with hub return, will we need to fix this?

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java [hunk 5]
```diff
@@ -110,6 +138,11 @@ public DaTaskView markDropCollected(UUID daId, UUID taskId) {
             task.setStartedAt(Instant.now());
             DaTaskView view = save(task);
             daEventProducer.emitDropCollected(daId, task.getCityId(), task.getShipmentId());
+            // M8-SEAM: in HUB_RETURN cities the DA collects the delivery FROM the hub, so record the
+            // hub-dest custody scan (ledger only — the DA's later DROP_* events drive the state).
+            if (isHubReturn(task)) {
+                hubScanSeamProducer.hubDestOut(task.getShipmentId());
```
Comment (Sidarthapati) [inline dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: also name the function to emithubdestout or something , convention to be followed?

File: dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java (hunk 2 above)
Comment (Sidarthapati) [reply dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java:89]: You're right, and no — the *real* dispatch delivery path does **not** use a centroid. `ShipmentEventsConsumer.handleStateChanged` (the `HANDED_TO_DROP_VAN` -> `assignDelivery` path every city uses today) passes the **actual** `event.getDestLat()/getDestLon()` from the shipment's drop address. This consumer falls back to a hex centroid only because M7's `ParcelSortedForDeliveryEvent` carries `destinationHexId` but not the real drop coords. The hex is exactly right for *which DA/territory* gets the parcel, but the centroid is a poor proxy for the *task location* the DA's queue sequences on.

Fix, two options: **(a)** enrich `ParcelSortedForDeliveryEvent` with `destLat/destLon` (M7 producer change, cleanest contract-wise), or **(b)** since `dispatch` already depends on `orders`, resolve the real drop coords from the shipment via an orders port here before calling `assignDelivery`. Leaning (b) so we don't touch M7's contract — will change unless you prefer (a).

File: dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java (hunk 1 above)
Comment (Sidarthapati) [reply dispatch/src/main/java/com/oneday/dispatch/events/HubDeliveryFeedConsumer.java:98]: Same point as the thread above: centroid is a fallback because the M7 event only carries the dest **hex**, not the real lat/lon. The genuine dispatch path (`ShipmentEventsConsumer`) uses the actual `destLat/destLon`. Will resolve the real drop coords (via an orders port) before `assignDelivery` instead of the hex mean.

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java (hunk 3 above)
Comment (Sidarthapati) [reply dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: Agreed — emitting `VAN_HANDOFF_COMPLETED` in a no-van city is semantically false. I reused the van-handoff method because it's structurally identical (DA reaches the rendezvous, hands parcels off) and the custody chain still lands correctly (`VAN_HANDOFF_COMPLETED` -> M4 `HANDED_TO_PICKUP_VAN`, then the `HUB_ORIGIN_IN` scan -> `AT_ORIGIN_HUB`). But the *event name* claiming "van" is wrong and would mislead any consumer keying on "a van received these."

Fix: make the emit mode-aware — in `HUB_RETURN` emit a neutral handoff event (e.g. `PICKUP_HANDED_OFF`) mapping to the same `HANDED_TO_PICKUP_VAN` -> `AT_ORIGIN_HUB` progression. The reused method is fine internally; only the emitted event should stop saying "van."

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java (hunk 4 above)
Comment (Sidarthapati) [reply dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: This line is actually the thing that *replaces* the van in a hub-return city: it's the M8 origin-hub inbound scan (`HUB_ORIGIN_IN`) fired because the DA physically drops the collected pickups **at the hub**, which advances the shipment to `AT_ORIGIN_HUB` — the correct end state. So this statement stays. The part that's wrong is the `emitVanHandoffCompleted` above it (see that thread): we'll swap that for a neutral hub-return handoff event, and this seam is exactly what stands in for the (non-existent) van.

File: dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java (hunk 5 above)
Comment (Sidarthapati) [reply dispatch/src/main/java/com/oneday/dispatch/service/impl/DaTaskServiceImpl.java]: Good catch — the producer's other methods are all `emit*` (`emitDropCollected`, `emitVanHandoffCompleted`, ...). Will rename `HubScanSeamProducer.hubOriginIn/hubDestOut` -> `emitHubOriginIn/emitHubDestOut` to match the convention.

File: orders/src/main/java/com/oneday/orders/events/DaEventsConsumer.java [hunk 6]
```diff
@@ -45,6 +45,8 @@ public void onDaEvent(DaLifecycleEvent event) {
             case PICKUP_ASSIGNED       -> ShipmentState.PICKUP_ASSIGNED;
             case PICKUP_FAILED         -> ShipmentState.PICKUP_FAILED;
             case VAN_HANDOFF_COMPLETED -> ShipmentState.HANDED_TO_PICKUP_VAN;
+            // HUB_RETURN city: DA dropped at the hub (no van) — same custody step as a van handoff.
+            case HUB_RETURN_HANDOFF_COMPLETED -> ShipmentState.HANDED_TO_PICKUP_VAN;
```
Comment (Sidarthapati) [inline orders/src/main/java/com/oneday/orders/events/DaEventsConsumer.java]: This shouldnt be handed to pickup van right?
